exporter/otlpexporter/config_test.go

```diff
@@ -134,6 +134,17 @@ func TestUnmarshalInvalidConfig(t *testing.T) {
 func TestValidDNSEndpoint(t *testing.T) {
 	factory := NewFactory()
 	cfg := factory.CreateDefaultConfig().(*Config)
-	cfg.Endpoint = "dns:///backend.example.com:4317"
+	cfg.Endpoint = "dns://backend.example.com:4317"
```

Comment:
Is this a valid dns endpoint? I though it would need either `///` in a row or another `/` somewhere to separate the authority and the path

Reply:
it's a good point, according to the grpc docs, the following is valid:

`dns:[//authority/]host[:port] -- DNS (default)`

I don't know how much more validation we should be introducing here

Reply:
The new string looks invalid to me, I'm surprised an error is not returned. I'm interpreting the regex as "if `//` is present, there must be a third `/`". What am I missing

Reply:
let me add a test for this case. i agree that it should return an error

Reply:
Oh I see how this test works now.  Currently the validation only checks that there is a port and that it is a valid integer, nothing else.  We should add full dns endpoint validation, but since the current functionality breaks users ability to use authorities in their dns endpoint I think we should add that later and merge this now.

For the sake of having a proper endpoint:

```suggestion
	cfg.Endpoint = "dns://authority/backend.example.com:4317"
```

Reply:
created https://github.com/open-telemetry/opentelemetry-collector/issues/10488